### ISSUE TITLE
When update the pacages with erros.

### ISSUE THREAD

**Issue by Aijunly** [JuliaLang/julia#53690]
I don't known why with the following errors.
```julia
(@v1.10) pkg> up
    Updating registry at `~/JuliaWorks/JuliaDepoRoll/registries/General`
    Updating git-repo `https://github.com/JuliaRegistries/General.git`
    Fetching: [===>                                     ]  6.1 %

ERROR: ArgumentError: invalid value for Enum Code: -30848
Stacktrace:
  [1] enum_argument_error(typename::Symbol, x::Int32)
    @ Base.Enums ./Enums.jl:93
  [2] Code
    @ ./Enums.jl:211 [inlined]
  [3] LibGit2.Error.GitError(err_code::Int32)
    @ LibGit2.Error /opt/juliaRoll/share/julia/stdlib/v1.10/LibGit2/src/error.jl:100
  [4] macro expansion
    @ /opt/juliaRoll/share/julia/stdlib/v1.10/LibGit2/src/error.jl:112 [inlined]
  [5] fetch(rmt::LibGit2.GitRemote, refspecs::Vector{String}; options::LibGit2.FetchOptions, msg::String)
    @ LibGit2 /opt/juliaRoll/share/julia/stdlib/v1.10/LibGit2/src/remote.jl:299
  [6] fetch
    @ /opt/juliaRoll/share/julia/stdlib/v1.10/LibGit2/src/remote.jl:294 [inlined]
  [7] fetch(repo::LibGit2.GitRepo; remote::String, remoteurl::SubString{String}, refspecs::Vector{String}, credentials::Nothing, callbacks::Dict{Symbol, Tuple{Ptr{Nothing}, Any}})
    @ LibGit2 /opt/juliaRoll/share/julia/stdlib/v1.10/LibGit2/src/LibGit2.jl:295
  [8] fetch(io::Base.TTY, repo::LibGit2.GitRepo, remoteurl::Nothing; header::Nothing, credentials::Nothing, refspecs::Vector{String}, kwargs::@Kwargs{})
    @ Pkg.GitTools /opt/juliaRoll/share/julia/stdlib/v1.10/Pkg/src/GitTools.jl:178
  [9] fetch (repeats 2 times)
    @ /opt/juliaRoll/share/julia/stdlib/v1.10/Pkg/src/GitTools.jl:140 [inlined]
 [10] (::Pkg.Registry.var"#77#82"{Pkg.Registry.RegistryInstance, Vector{Tuple{String, String}}, String, Base.TTY, Dict{String, Any}})(repo::LibGit2.GitRepo)
    @ Pkg.Registry /opt/juliaRoll/share/julia/stdlib/v1.10/Pkg/src/Registry/Registry.jl:464
 [11] with(f::Pkg.Registry.var"#77#82"{Pkg.Registry.RegistryInstance, Vector{Tuple{String, String}}, String, Base.TTY, Dict{String, Any}}, obj::LibGit2.GitRepo)
    @ LibGit2 /opt/juliaRoll/share/julia/stdlib/v1.10/LibGit2/src/types.jl:1160
 [12] (::Pkg.Registry.var"#73#78"{Base.TTY, Dates.Second, Dict{String, Any}, String, Vector{Pkg.Registry.RegistryInstance}})()
    @ Pkg.Registry /opt/juliaRoll/share/julia/stdlib/v1.10/Pkg/src/Registry/Registry.jl:449
 [13] mkpidlock(f::Pkg.Registry.var"#73#78"{Base.TTY, Dates.Second, Dict{String, Any}, String, Vector{Pkg.Registry.RegistryInstance}}, at::String, pid::Int32; kwopts::@Kwargs{stale_age::Int64})
    @ FileWatching.Pidfile /opt/juliaRoll/share/julia/stdlib/v1.10/FileWatching/src/pidfile.jl:93
 [14] mkpidlock
    @ /opt/juliaRoll/share/julia/stdlib/v1.10/FileWatching/src/pidfile.jl:88 [inlined]
 [15] update(regs::Vector{Pkg.Registry.RegistrySpec}; io::Base.TTY, force::Bool, depots::Vector{String}, update_cooldown::Dates.Second)
    @ Pkg.Registry /opt/juliaRoll/share/julia/stdlib/v1.10/Pkg/src/Registry/Registry.jl:379
 [16] update
    @ /opt/juliaRoll/share/julia/stdlib/v1.10/Pkg/src/Registry/Registry.jl:372 [inlined]
 [17] update_registries(ctx::Pkg.Types.Context; force::Bool, kwargs::@Kwargs{})
    @ Pkg.Operations /opt/juliaRoll/share/julia/stdlib/v1.10/Pkg/src/Operations.jl:1269
 [18] update_registries
    @ /opt/juliaRoll/share/julia/stdlib/v1.10/Pkg/src/Operations.jl:1266 [inlined]
 [19] up(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec}; level::Pkg.Types.UpgradeLevel, mode::Pkg.Types.PackageMode, preserve::Nothing, update_registry::Bool, skip_writing_project::Bool, kwargs::@Kwargs{io::Base.TTY})
    @ Pkg.API /opt/juliaRoll/share/julia/stdlib/v1.10/Pkg/src/API.jl:339
 [20] up(pkgs::Vector{Pkg.Types.PackageSpec}; io::Base.TTY, kwargs::@Kwargs{})
    @ Pkg.API /opt/juliaRoll/share/julia/stdlib/v1.10/Pkg/src/API.jl:159
 [21] up(pkgs::Vector{Pkg.Types.PackageSpec})
    @ Pkg.API /opt/juliaRoll/share/julia/stdlib/v1.10/Pkg/src/API.jl:148
 [22] do_cmd!(command::Pkg.REPLMode.Command, repl::REPL.LineEditREPL)
    @ Pkg.REPLMode /opt/juliaRoll/share/julia/stdlib/v1.10/Pkg/src/REPLMode/REPLMode.jl:412
 [23] do_cmd(repl::REPL.LineEditREPL, input::String; do_rethrow::Bool)
    @ Pkg.REPLMode /opt/juliaRoll/share/julia/stdlib/v1.10/Pkg/src/REPLMode/REPLMode.jl:390
 [24] do_cmd
    @ /opt/juliaRoll/share/julia/stdlib/v1.10/Pkg/src/REPLMode/REPLMode.jl:380 [inlined]
 [25] (::Pkg.REPLMode.var"#24#27"{REPL.LineEditREPL, REPL.LineEdit.Prompt})(s::REPL.LineEdit.MIState, buf::IOBuffer, ok::Bool)
    @ Pkg.REPLMode /opt/juliaRoll/share/julia/stdlib/v1.10/Pkg/src/REPLMode/REPLMode.jl:557
 [26] #invokelatest#2
    @ ./essentials.jl:892 [inlined]
 [27] invokelatest
    @ ./essentials.jl:889 [inlined]
 [28] run_interface(terminal::REPL.Terminals.TextTerminal, m::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
    @ REPL.LineEdit /opt/juliaRoll/share/julia/stdlib/v1.10/REPL/src/LineEdit.jl:2656
 [29] run_frontend(repl::REPL.LineEditREPL, backend::REPL.REPLBackendRef)
    @ REPL /opt/juliaRoll/share/julia/stdlib/v1.10/REPL/src/REPL.jl:1312
 [30] (::REPL.var"#62#68"{REPL.LineEditREPL, REPL.REPLBackendRef})()
    @ REPL /opt/juliaRoll/share/julia/stdlib/v1.10/REPL/src/REPL.jl:386
```

**Comment by Seelengrab** [comment]
Is there a Project.toml+Manifest.toml that can reproduce this?

**Comment by KristofferC** [comment]
I doubt the Project + Manifest has anything to do with this since it is just updating a registry.

More relevant, how was Julia installed?

**Comment by Seelengrab** [comment]
Pure guesswork, but from the filepaths it seems to be a cluster-handled/managed install:

```
/opt/juliaRoll/share/julia/stdlib/v1.10/LibGit2/src/error.jl:100
```

So my gut says this has to do with a mismatch between the Julia-shipped `LibGit2` and the one provided by the host?

**Comment by mbauman** [comment]
That might be the case.  Or it's an unexpected network problem that leads to LibGit2 unexpectedly returning returning what [looks to be an mbedtls error code](https://duckduckgo.com/?q=error+0x7880) from one of its functions without marshaling it.

**Comment by JunWang-JUST** [comment]
Thanks very much.

I manually install Julia by the followings commands : decompress julia.tar.gz, rename it with juliaRoll and move juliaRoll /opt/.

[Manifest.toml.zip](https://github.com/JuliaLang/julia/files/14582030/Manifest.toml.zip)

**Comment by JunWang-JUST** [comment]
$ uname -a
Linux malegbasy 6.1.0-13-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.55-1 (2023-09-29) x86_64 GNU/Linux

**Comment by fingolfin** [comment]
Missing information from submitter, and we can't reproduce it -> closing.

**Comment by maleadt** [comment]
FWIW, I can reproduce something like this on 1.10 with a particular Julia clone on `cyclops`:

```julia
using LibGit2

# a tarball of the particular cloned repository
const TARBALL = joinpath(@__DIR__, "julia.tar")

tmpdir = mktempdir()
run(`tar -xf $TARBALL -C $tmpdir`)
repo = LibGit2.GitRepo(joinpath(tmpdir, "julia"))

try
    remote = LibGit2.lookup_remote(repo, "origin")
    try
        LibGit2.fetch(remote, ["+refs/heads/master:refs/heads/master"])
        println("Fetch succeeded!")
    finally
        close(remote)
    end
finally
    close(repo)
    rm(tmpdir; recursive=true, force=true)
end
```

```
ERROR: LoadError: ArgumentError: invalid value for Enum Code: -30848
Stacktrace:
 [1] enum_argument_error(typename::Symbol, x::Int32)
   @ Base.Enums ./Enums.jl:93
 [2] Code
   @ ./Enums.jl:212 [inlined]
 [3] LibGit2.Error.GitError(err_code::Int32)
   @ LibGit2.Error ~/Julia/depot/juliaup/julia-1.10.10+0.x64.linux.gnu/share/julia/stdlib/v1.10/LibGit2/src/error.jl:100
 [4] macro expansion
   @ ~/Julia/depot/juliaup/julia-1.10.10+0.x64.linux.gnu/share/julia/stdlib/v1.10/LibGit2/src/error.jl:112 [inlined]
 [5] fetch(rmt::LibGit2.GitRemote, refspecs::Vector{String}; options::LibGit2.FetchOptions, msg::String)
   @ LibGit2 ~/Julia/depot/juliaup/julia-1.10.10+0.x64.linux.gnu/share/julia/stdlib/v1.10/LibGit2/src/remote.jl:299
 [6] fetch(rmt::LibGit2.GitRemote, refspecs::Vector{String})
   @ LibGit2 ~/Julia/depot/juliaup/julia-1.10.10+0.x64.linux.gnu/share/julia/stdlib/v1.10/LibGit2/src/remote.jl:294
 [7] top-level scope
   @ ~/Julia/src/julia/reproducer_min.jl:15
```

This does not reproduce on a different system, even with `JULIA_SSL_CA_ROOTS_PATH` set or `LibGit2.set_ssl_cert_locations` called to point to the problematic system's `ca-certificates.crt`.

It also does not reproduce on 1.11, so I'm trying to bisect what changed (likely a `libgit2` update, or something related to NetworkOptions.jl).

In the meantime, if anybody has other suggestions on which part of the system could influence this, or wants to try this out on `cyclops`, please let me know.

**Comment by maleadt** [comment]
Fix bisected to a4b5ad3d4c6c56056ac1fd55d012ca2a6d234d35. Backporting https://github.com/JuliaLang/julia/pull/51259 to an LTS seems iffy though...
